client/mm/libxc/binance.go

```diff
@@ -425,17 +426,43 @@ func newBinance(apiKey, secretKey string, log dex.Logger, net dex.Network, binan
 	return bnc
 }
 
-// setBalances stores the balances of the user.
-func (bnc *binance) setBalances(coinsData []*binanceCoinInfo) {
+// setBalances queries binance for the user's balances and stores them in the
+// balances map.
+func (bnc *binance) setBalances(ctx context.Context) error {
+	var resp struct {
+		Balances []struct {
+			Asset  string  `json:"asset"`
+			Free   float64 `json:"free,string"`
+			Locked float64 `json:"locked,string"`
+		} `json:"balances"`
+	}
+	err := bnc.getAPI(ctx, "/api/v3/account", nil, true, true, &resp)
```

Comment:
Hmm. I made some changes in #2615 that might be relevant to these changes too. I think we should just expect to be connected to configured cexes, or connect on-the-fly when we need info.

Reply:
This change just updates the API used to get the balances. I think your changes over there are still good.

> I think we should just expect to be connected to configured cexes, or connect on-the-fly when we need info.

I think the use should make some action before connecting the the CEX. It shouldn't automatically connect at start-up even if the user doesn't intend to use market making at all. They may want to set up some configurations before connecting.